### PR TITLE
Avoid hardwired version for GM in startup script and bundle dir

### PR DESCRIPTION
**Pull request by PaulWessel**
GraphicsMagick has a support dir with version number in it (GraphicsMagick-13.33) which obviously will change in the future.  This PR avoids that by placing the files (from whatever version) into a version-free directory name.
Also let CMake set variables *_VERSION for gm, gs, and ffmpeg as these may come in handy.

### CODE REVIEW

**Comment by PaulWessel** [comment]
@seisman, when you double-click on the bundle logo, is your terminal width too wide? The math I did was done at work on a 4k display and there are two possible issues:

I have a 43" 4k TV as monitor
I have set the display fonts to something large (18p or 24p)

So I am not sure if my calculations to scale the 80x24 terminal size is general enough.  Works for me there, but at home the terminal is way too wide.

**Comment by PaulWessel** [comment]
Given that the important GM files are the ones pointed to by MAGICK_CONFIGURE_PATH , perhaps we do not need to place these few files in share at all (also with dumb version number):

ls share/GraphicsMagick-1.3.33/config/
colors.mgk	log.mgk		modules.mgk

Both the colors and modules files say they are optional and that GM will run without them, but that message is not in log.mgk.

Googling these says that GraphicsMagick uses a number of XML format configuration files, and it lists the log.mgk as well as type.mgk which we have in the other directory in lib.  Looks messed up on their end.  I wonder if we should add those three files to the share/GraphicsMagick dir.  Of maybe this looks different on brew.  Do you have two Graphicsmagick-1.3.33 directories (lib and share)?

**Comment by seisman** [comment]
It's more complicated for homebrew. There are config files in both lib and share directory, the same as macports. But there are more files under the lib directory.

These are the files in `/usr/local/Cellar/graphicsmagick/1.3.33/lib/GraphicsMagick`.
```
.
├── config
│   ├── delegates.mgk
│   ├── type-ghostscript.mgk
│   ├── type-solaris.mgk
│   ├── type-windows.mgk
│   └── type.mgk
└── modules-Q16
    ├── coders
    │   ├── art.la
    │   ├── art.so
    │   ├── avs.la
    │   ├── avs.so
    │   ├── bmp.la
    │   ├── bmp.so
    │   ├── braille.la
    │   ├── braille.so
    │   ├── cals.la
    │   ├── cals.so
    │   ├── caption.la
    │   ├── caption.so
    │   ├── cineon.la
    │   ├── cineon.so
    │   ├── cmyk.la
    │   ├── cmyk.so
    │   ├── cut.la
    │   ├── cut.so
    │   ├── dcm.la
    │   ├── dcm.so
    │   ├── dcraw.la
    │   ├── dcraw.so
    │   ├── dib.la
    │   ├── dib.so
    │   ├── dpx.la
    │   ├── dpx.so
    │   ├── ept.la
    │   ├── ept.so
    │   ├── fax.la
    │   ├── fax.so
    │   ├── fits.la
    │   ├── fits.so
    │   ├── gif.la
    │   ├── gif.so
    │   ├── gradient.la
    │   ├── gradient.so
    │   ├── gray.la
    │   ├── gray.so
    │   ├── histogram.la
    │   ├── histogram.so
    │   ├── hrz.la
    │   ├── hrz.so
    │   ├── html.la
    │   ├── html.so
    │   ├── icon.la
    │   ├── icon.so
    │   ├── identity.la
    │   ├── identity.so
    │   ├── info.la
    │   ├── info.so
    │   ├── jnx.la
    │   ├── jnx.so
    │   ├── jp2.la
    │   ├── jp2.so
    │   ├── jpeg.la
    │   ├── jpeg.so
    │   ├── label.la
    │   ├── label.so
    │   ├── locale.la
    │   ├── locale.so
    │   ├── logo.la
    │   ├── logo.so
    │   ├── mac.la
    │   ├── mac.so
    │   ├── map.la
    │   ├── map.so
    │   ├── mat.la
    │   ├── mat.so
    │   ├── matte.la
    │   ├── matte.so
    │   ├── meta.la
    │   ├── meta.so
    │   ├── miff.la
    │   ├── miff.so
    │   ├── mono.la
    │   ├── mono.so
    │   ├── mpc.la
    │   ├── mpc.so
    │   ├── mpeg.la
    │   ├── mpeg.so
    │   ├── mpr.la
    │   ├── mpr.so
    │   ├── msl.la
    │   ├── msl.so
    │   ├── mtv.la
    │   ├── mtv.so
    │   ├── mvg.la
    │   ├── mvg.so
    │   ├── null.la
    │   ├── null.so
    │   ├── otb.la
    │   ├── otb.so
    │   ├── palm.la
    │   ├── palm.so
    │   ├── pcd.la
    │   ├── pcd.so
    │   ├── pcl.la
    │   ├── pcl.so
    │   ├── pcx.la
    │   ├── pcx.so
    │   ├── pdb.la
    │   ├── pdb.so
    │   ├── pdf.la
    │   ├── pdf.so
    │   ├── pict.la
    │   ├── pict.so
    │   ├── pix.la
    │   ├── pix.so
    │   ├── plasma.la
    │   ├── plasma.so
    │   ├── png.la
    │   ├── png.so
    │   ├── pnm.la
    │   ├── pnm.so
    │   ├── preview.la
    │   ├── preview.so
    │   ├── ps.la
    │   ├── ps.so
    │   ├── ps2.la
    │   ├── ps2.so
    │   ├── ps3.la
    │   ├── ps3.so
    │   ├── pwp.la
    │   ├── pwp.so
    │   ├── rgb.la
    │   ├── rgb.so
    │   ├── rla.la
    │   ├── rla.so
    │   ├── rle.la
    │   ├── rle.so
    │   ├── sct.la
    │   ├── sct.so
    │   ├── sfw.la
    │   ├── sfw.so
    │   ├── sgi.la
    │   ├── sgi.so
    │   ├── stegano.la
    │   ├── stegano.so
    │   ├── sun.la
    │   ├── sun.so
    │   ├── svg.la
    │   ├── svg.so
    │   ├── tga.la
    │   ├── tga.so
    │   ├── tiff.la
    │   ├── tiff.so
    │   ├── tile.la
    │   ├── tile.so
    │   ├── tim.la
    │   ├── tim.so
    │   ├── topol.la
    │   ├── topol.so
    │   ├── ttf.la
    │   ├── ttf.so
    │   ├── txt.la
    │   ├── txt.so
    │   ├── uil.la
    │   ├── uil.so
    │   ├── url.la
    │   ├── url.so
    │   ├── uyvy.la
    │   ├── uyvy.so
    │   ├── vicar.la
    │   ├── vicar.so
    │   ├── vid.la
    │   ├── vid.so
    │   ├── viff.la
    │   ├── viff.so
    │   ├── wbmp.la
    │   ├── wbmp.so
    │   ├── webp.la
    │   ├── webp.so
    │   ├── wmf.la
    │   ├── wmf.so
    │   ├── wpg.la
    │   ├── wpg.so
    │   ├── xbm.la
    │   ├── xbm.so
    │   ├── xc.la
    │   ├── xc.so
    │   ├── xcf.la
    │   ├── xcf.so
    │   ├── xpm.la
    │   ├── xpm.so
    │   ├── yuv.la
    │   └── yuv.so
    └── filters
        ├── analyze.la
        └── analyze.so

4 directories, 193 files

```

**Comment by PaulWessel** [comment]
OK, so no log.mgk for you either.  I think the MacPorts share dir with those three mgk files can go. At the very least, we could copy log.mkg over to that directory and be done with it if we worry about it.

**Comment by seisman** [comment]
Yes, I agree.

**Comment by PaulWessel** [comment]
Any suggestion on how to place all the files (there are no subdirectories) from /opt/local/lib/proj6/share/proj into share/proj6 without creating the symbolic link /opt/local/share/proj6?

Hm, I could probably create that link in the build directory instead and remove it when done?

**Comment by seisman** [comment]
Does this work?
```
install (DIRECTORY
        /opt/local/lib/proj6/share/proj
	DESTINATION \${GMT_SHAREDIR}
	COMPONENT Runtime)
```

File: admin/build-macos-external-list.sh [hunk 1]
```diff
@@ -93,13 +93,20 @@ if [ ! "X$EXESHARED" = "X" ]; then
 fi
 cat << EOF
 
+# Place the licenses for runtime dependencies
 install (DIRECTORY
 	../../admin/Licenses
 	DESTINATION share
```
Comment (seisman) [inline admin/build-macos-external-list.sh]: Does this change looks better?
```suggestion
	DESTINATION \${GMT_DATADIR}
```

Comment (PaulWessel) [reply admin/build-macos-external-list.sh]: see below

**Comment by PaulWessel** [comment]
yes in a way.  I decided it is just silly to insist that directory must be called proj6 since we set PROJ_DIR anyway.  So now I just place the proj dir in share and changed the startup_macos script.

**Comment by PaulWessel** [comment]
This is better - no crazy duplication of MacPorts files in the system dirs.

**Comment by PaulWessel** [comment]
Any comment on the terminal width, @seisman ?

**Comment by seisman** [comment]
> Any comment on the terminal width, @seisman ?

I have no better idea.

**Comment by PaulWessel** [comment]
OK, closing this for now then.  The build-release.sh got simplified quite a bit.  Once gs 9.50 appears in MacPorts then another bit can be removed.

**Comment by seisman** [comment]
Why don't merge it?

**Comment by PaulWessel** [comment]
sorry, done.

**Pull request merged**